DevOps.Util/HelixServer.cs

```diff
@@ -19,21 +19,21 @@ public HelixServer()
             _client = new DevOpsHttpClient();
         }
 
-        private IHelixApi HelixApi { get; } = ApiFactory.GetAnonymous();
-
-        public async ValueTask GetHelixPayloads(string jobId, List<string> workItems, string downloadDir)
+        public async ValueTask GetHelixPayloads(string jobId, List<string> workItems, string downloadDir, string? token)
```

Comment:
Rather than use a token per request why not store the token in `HelixServer`? Won't we always end up passing the same token for every call?

Reply:
Technically yes, but what if I wanted to get an anonymous job and an internal job on the same runfo session? I would have to try what would happen if I try to get an anonymous job with an authenticated HelixApi.

Reply:
In either case though at the callsite you're going to have to change the invocation. Essentially at the call site you have to make a determination: do I need or not need authentication here? Based on that you will have to change the API or the server that you use to make the call. 

Don't have a strong preference here but in the other cases we did the authentication at the server level.

Reply:
Makes sense. Then I'll add caching of the token. I'll follow what we do with the azure devops token.

Reply:
Done.